pkg/k8s/init.go

```diff
@@ -18,14 +18,72 @@ package k8s
 import (
 	"fmt"
 	"os"
+	"time"
 
+	"github.com/cilium/cilium/pkg/backoff"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
 
 	go_version "github.com/hashicorp/go-version"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	nodeRetrievalMaxRetries = 15
+)
+
+func waitForNodeInformation(nodeName string) *node.Node {
+	backoff := backoff.Exponential{
+		Min:    time.Duration(200) * time.Millisecond,
+		Factor: 2.0,
+		Name:   "k8s-node-retrieval",
+	}
+
+	for retry := 0; retry < nodeRetrievalMaxRetries; retry++ {
+		n, err := retrieveNodeInformation(nodeName)
+		if err != nil {
+			log.WithError(err).Warning("Waiting for k8s node information")
+			backoff.Wait()
+			continue
+		}
+
+		return n
+	}
+
+	return nil
+}
+
+func retrieveNodeInformation(nodeName string) (*node.Node, error) {
+	requireIPv4CIDR := option.Config.K8sRequireIPv4PodCIDR
+	requireIPv6CIDR := option.Config.K8sRequireIPv6PodCIDR
+
+	k8sNode, err := GetNode(Client(), nodeName)
+	if err != nil {
+		// If no CIDR is required, retrieving the node information is
+		// optional
+		if !requireIPv4CIDR && !requireIPv6CIDR {
```

Comment:
Why not check this first instead of doing `GetNode()`?

Reply:
I'm not sure I understand the comment. The intent of the code is to do GetNode() in a best effort approach unless the CIDR is required in which case it should return failure if the node resource cannot be retrieved.

Reply:
Got it, thanks